### ISSUE TITLE
pub build option to include dart files

### ISSUE THREAD

**Issue by DartBot** [dart-lang/pub#706]
<a href="https://github.com/rbishop-bah"><img src="https://avatars.githubusercontent.com/u/3227777?v=3" align="left" width="96" height="96"hspace="10"></img></a> **Issue by [rbishop-bah](https://github.com/rbishop-bah)**
_Originally opened as dart-lang/sdk#14792_

----

pub build currently leaves out all of the dart artifacts and supports the JS deployment case only.

While we wait for dart2dart to mature, it'd be a useful enhancement to allow pub build to output (into the /build folder) both the JavaScript and Dart pieces and have the index.html use the right ones depending on the browser in which it's run (like we had with web_ui).

This would allow construction of WARs that can be deployed once and run in all browsers including Dartium.  This is often very useful for debugging problems.

If I understand correctly, that is an eventual goal of pub build once dart2dart is ready.  But using the unsquashed dart right now would really help.

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

The correct way to do this for now is probably to have the removal of the Dart files be based on the mode, and to only remove them in deploy mode. That will make "pub build --mode=debug" preserve the Dart files.
______
*Added this to the [Later](../milestones/Later) milestone.*
*Added [Area-Pub](../labels/Area-Pub), [Triaged](../labels/Triaged) labels.*

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

Issue dart-lang/sdk#15000 has been merged into this issue.

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

*Set owner to @munificent.*
*Added [Started](../labels/Started) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

*Added [Fixed](../labels/Fixed) label.*
